src/erlcloud_firehose_impl.erl

```diff
@@ -0,0 +1,177 @@
+%% -*- mode: erlang;erlang-indent-level: 4;indent-tabs-mode: nil -*-
```

Comment:
this seems to be duplicate from `erlcloud_kinesis_impl`, can we not reuse that implementation here?